### ISSUE TITLE
Integration tests lack asserts

### ISSUE THREAD

**Issue by AnkBurov** [maxpumperla/elephas#129]
Hi. I've noticed that your 'integration' tests (that do end to end tests like test_spark_model_end_to_end) do not have any asserts. 

Is this a bug or a feature?

**Comment by maxpumperla** [comment]
@AnkBurov this is meant as a safeguard. When, e.g. Keras changes its API, I want to see my examples fail. Unit tests often don't cover that. and I want it automated. So even if there are no asserts, those tests can still fail (and did a few times in the past).

Btw, It's not obvious what a good IT suite for ML software looks like. What do you want me to assert? That accuracy is higher than before training? Looking forward to suggestions

**Comment by AnkBurov** [comment]
Okay, thanks for the explanation of the reasons.
> What do you want me to assert?
Usually replacing print() with assert is a good thing to start from. For example
```python
# sync batch
    spark_model = SparkModel(model, frequency='batch',
                             mode='synchronous', num_workers=2)
    spark_model.fit(rdd, epochs=epochs, batch_size=batch_size,
                    verbose=2, validation_split=0.1)
    score = spark_model.master_network.evaluate(x_test, y_test, verbose=2)
    print('Test accuracy:', score[1])
```
asserting model that test accuracy is higher than some reasonable value would have prevented this issue: https://github.com/maxpumperla/elephas/issues/126
Not that I care much, but just sayin'.

**Comment by maxpumperla** [comment]
@AnkBurov thanks for repeating my suggestion and stating your indifference. there are always things to improve, so I'm definitely interested in having conversations with people who do care and actually want to help. Talk is cheap and pointing out flaws in other projects is certainly easier than creating something meaningful yourself, as you might learn at some point. Take care!
